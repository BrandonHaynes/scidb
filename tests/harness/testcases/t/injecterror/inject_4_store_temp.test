# TEST:    Send a signal 11 to just the child process of a worker instance
# EXPECT:  After scidb recovers, temp arrays should be unavailable for queries,
#          but still part of the catalog.

--setup

load_library('misc')

create temp array ARRAY1 <v:int64> [I=0:100,1,0]
create temp array ARRAY2 <v:int64> [I=0:100,1,0]
create      array ARRAY3 <v:int64> [I=0:100,1,0]
store (build (ARRAY1, I), ARRAY1)
store (build (ARRAY2,-I), ARRAY2)
store (build (ARRAY3,-I), ARRAY3)

--test
--start-query-logging

# check distribution
aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=0),count(*))
aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=1),count(*))
aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=2),count(*))
aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=3),count(*))

# inject error into a write query
apply(ARRAY1,zz,injectError(0, 4))
--error --code=scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR "store(build(ARRAY1,0),ARRAY1)"

# check that the data in the arrays is not altered:
aggregate(filter(apply(join(ARRAY1,ARRAY2),attrSum,ARRAY1.v + ARRAY2.v), attrSum<>0),count(*))

# inject error into a write query
apply(ARRAY1,zz,injectError(3, 4))
--error --code=scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR "store(ARRAY3,ARRAY1)"

# check that the data in the arrays is not altered:
aggregate(filter(apply(join(ARRAY1,ARRAY2),attrSum,ARRAY1.v + ARRAY2.v), attrSum<>0),count(*))

--cleanup
remove(ARRAY1)
remove(ARRAY2)
remove(ARRAY3)
