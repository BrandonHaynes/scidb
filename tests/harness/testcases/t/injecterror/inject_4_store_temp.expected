SCIDB QUERY : <load_library('misc')>
Query was executed successfully

SCIDB QUERY : <create temp array ARRAY1 <v:int64> [I=0:100,1,0]>
Query was executed successfully

SCIDB QUERY : <create temp array ARRAY2 <v:int64> [I=0:100,1,0]>
Query was executed successfully

SCIDB QUERY : <create      array ARRAY3 <v:int64> [I=0:100,1,0]>
Query was executed successfully

SCIDB QUERY : <store (build (ARRAY1, I), ARRAY1)>
{I} v
{0} 0
{1} 1
{2} 2
{3} 3
{4} 4
{5} 5
{6} 6
{7} 7
{8} 8
{9} 9
{10} 10
{11} 11
{12} 12
{13} 13
{14} 14
{15} 15
{16} 16
{17} 17
{18} 18
{19} 19
{20} 20
{21} 21
{22} 22
{23} 23
{24} 24
{25} 25
{26} 26
{27} 27
{28} 28
{29} 29
{30} 30
{31} 31
{32} 32
{33} 33
{34} 34
{35} 35
{36} 36
{37} 37
{38} 38
{39} 39
{40} 40
{41} 41
{42} 42
{43} 43
{44} 44
{45} 45
{46} 46
{47} 47
{48} 48
{49} 49
{50} 50
{51} 51
{52} 52
{53} 53
{54} 54
{55} 55
{56} 56
{57} 57
{58} 58
{59} 59
{60} 60
{61} 61
{62} 62
{63} 63
{64} 64
{65} 65
{66} 66
{67} 67
{68} 68
{69} 69
{70} 70
{71} 71
{72} 72
{73} 73
{74} 74
{75} 75
{76} 76
{77} 77
{78} 78
{79} 79
{80} 80
{81} 81
{82} 82
{83} 83
{84} 84
{85} 85
{86} 86
{87} 87
{88} 88
{89} 89
{90} 90
{91} 91
{92} 92
{93} 93
{94} 94
{95} 95
{96} 96
{97} 97
{98} 98
{99} 99
{100} 100

SCIDB QUERY : <store (build (ARRAY2,-I), ARRAY2)>
{I} v
{0} 0
{1} -1
{2} -2
{3} -3
{4} -4
{5} -5
{6} -6
{7} -7
{8} -8
{9} -9
{10} -10
{11} -11
{12} -12
{13} -13
{14} -14
{15} -15
{16} -16
{17} -17
{18} -18
{19} -19
{20} -20
{21} -21
{22} -22
{23} -23
{24} -24
{25} -25
{26} -26
{27} -27
{28} -28
{29} -29
{30} -30
{31} -31
{32} -32
{33} -33
{34} -34
{35} -35
{36} -36
{37} -37
{38} -38
{39} -39
{40} -40
{41} -41
{42} -42
{43} -43
{44} -44
{45} -45
{46} -46
{47} -47
{48} -48
{49} -49
{50} -50
{51} -51
{52} -52
{53} -53
{54} -54
{55} -55
{56} -56
{57} -57
{58} -58
{59} -59
{60} -60
{61} -61
{62} -62
{63} -63
{64} -64
{65} -65
{66} -66
{67} -67
{68} -68
{69} -69
{70} -70
{71} -71
{72} -72
{73} -73
{74} -74
{75} -75
{76} -76
{77} -77
{78} -78
{79} -79
{80} -80
{81} -81
{82} -82
{83} -83
{84} -84
{85} -85
{86} -86
{87} -87
{88} -88
{89} -89
{90} -90
{91} -91
{92} -92
{93} -93
{94} -94
{95} -95
{96} -96
{97} -97
{98} -98
{99} -99
{100} -100

SCIDB QUERY : <store (build (ARRAY3,-I), ARRAY3)>
{I} v
{0} 0
{1} -1
{2} -2
{3} -3
{4} -4
{5} -5
{6} -6
{7} -7
{8} -8
{9} -9
{10} -10
{11} -11
{12} -12
{13} -13
{14} -14
{15} -15
{16} -16
{17} -17
{18} -18
{19} -19
{20} -20
{21} -21
{22} -22
{23} -23
{24} -24
{25} -25
{26} -26
{27} -27
{28} -28
{29} -29
{30} -30
{31} -31
{32} -32
{33} -33
{34} -34
{35} -35
{36} -36
{37} -37
{38} -38
{39} -39
{40} -40
{41} -41
{42} -42
{43} -43
{44} -44
{45} -45
{46} -46
{47} -47
{48} -48
{49} -49
{50} -50
{51} -51
{52} -52
{53} -53
{54} -54
{55} -55
{56} -56
{57} -57
{58} -58
{59} -59
{60} -60
{61} -61
{62} -62
{63} -63
{64} -64
{65} -65
{66} -66
{67} -67
{68} -68
{69} -69
{70} -70
{71} -71
{72} -72
{73} -73
{74} -74
{75} -75
{76} -76
{77} -77
{78} -78
{79} -79
{80} -80
{81} -81
{82} -82
{83} -83
{84} -84
{85} -85
{86} -86
{87} -87
{88} -88
{89} -89
{90} -90
{91} -91
{92} -92
{93} -93
{94} -94
{95} -95
{96} -96
{97} -97
{98} -98
{99} -99
{100} -100

SCIDB QUERY : <aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=0),count(*))>
{i} count
{0} 26

SCIDB QUERY : <aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=1),count(*))>
{i} count
{0} 25

SCIDB QUERY : <aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=2),count(*))>
{i} count
{0} 25

SCIDB QUERY : <aggregate(filter(apply(ARRAY1, instanceid, instanceid()), instanceid=3),count(*))>
{i} count
{0} 25

SCIDB QUERY : <apply(ARRAY1,zz,injectError(0, 4))>
{I} v,zz
{0} 0,4
{1} 1,-1
{2} 2,-1
{3} 3,-1
{4} 4,4
{5} 5,-1
{6} 6,-1
{7} 7,-1
{8} 8,4
{9} 9,-1
{10} 10,-1
{11} 11,-1
{12} 12,4
{13} 13,-1
{14} 14,-1
{15} 15,-1
{16} 16,4
{17} 17,-1
{18} 18,-1
{19} 19,-1
{20} 20,4
{21} 21,-1
{22} 22,-1
{23} 23,-1
{24} 24,4
{25} 25,-1
{26} 26,-1
{27} 27,-1
{28} 28,4
{29} 29,-1
{30} 30,-1
{31} 31,-1
{32} 32,4
{33} 33,-1
{34} 34,-1
{35} 35,-1
{36} 36,4
{37} 37,-1
{38} 38,-1
{39} 39,-1
{40} 40,4
{41} 41,-1
{42} 42,-1
{43} 43,-1
{44} 44,4
{45} 45,-1
{46} 46,-1
{47} 47,-1
{48} 48,4
{49} 49,-1
{50} 50,-1
{51} 51,-1
{52} 52,4
{53} 53,-1
{54} 54,-1
{55} 55,-1
{56} 56,4
{57} 57,-1
{58} 58,-1
{59} 59,-1
{60} 60,4
{61} 61,-1
{62} 62,-1
{63} 63,-1
{64} 64,4
{65} 65,-1
{66} 66,-1
{67} 67,-1
{68} 68,4
{69} 69,-1
{70} 70,-1
{71} 71,-1
{72} 72,4
{73} 73,-1
{74} 74,-1
{75} 75,-1
{76} 76,4
{77} 77,-1
{78} 78,-1
{79} 79,-1
{80} 80,4
{81} 81,-1
{82} 82,-1
{83} 83,-1
{84} 84,4
{85} 85,-1
{86} 86,-1
{87} 87,-1
{88} 88,4
{89} 89,-1
{90} 90,-1
{91} 91,-1
{92} 92,4
{93} 93,-1
{94} 94,-1
{95} 95,-1
{96} 96,4
{97} 97,-1
{98} 98,-1
{99} 99,-1
{100} 100,4

SCIDB QUERY : <store(build(ARRAY1,0),ARRAY1)>
[An error expected at this place for the query "store(build(ARRAY1,0),ARRAY1)". And it failed with error code = scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR. Expected error code = scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR.]

SCIDB QUERY : <aggregate(filter(apply(join(ARRAY1,ARRAY2),attrSum,ARRAY1.v + ARRAY2.v), attrSum<>0),count(*))>
{i} count
{0} 0

SCIDB QUERY : <apply(ARRAY1,zz,injectError(3, 4))>
{I} v,zz
{0} 0,-1
{1} 1,-1
{2} 2,-1
{3} 3,4
{4} 4,-1
{5} 5,-1
{6} 6,-1
{7} 7,4
{8} 8,-1
{9} 9,-1
{10} 10,-1
{11} 11,4
{12} 12,-1
{13} 13,-1
{14} 14,-1
{15} 15,4
{16} 16,-1
{17} 17,-1
{18} 18,-1
{19} 19,4
{20} 20,-1
{21} 21,-1
{22} 22,-1
{23} 23,4
{24} 24,-1
{25} 25,-1
{26} 26,-1
{27} 27,4
{28} 28,-1
{29} 29,-1
{30} 30,-1
{31} 31,4
{32} 32,-1
{33} 33,-1
{34} 34,-1
{35} 35,4
{36} 36,-1
{37} 37,-1
{38} 38,-1
{39} 39,4
{40} 40,-1
{41} 41,-1
{42} 42,-1
{43} 43,4
{44} 44,-1
{45} 45,-1
{46} 46,-1
{47} 47,4
{48} 48,-1
{49} 49,-1
{50} 50,-1
{51} 51,4
{52} 52,-1
{53} 53,-1
{54} 54,-1
{55} 55,4
{56} 56,-1
{57} 57,-1
{58} 58,-1
{59} 59,4
{60} 60,-1
{61} 61,-1
{62} 62,-1
{63} 63,4
{64} 64,-1
{65} 65,-1
{66} 66,-1
{67} 67,4
{68} 68,-1
{69} 69,-1
{70} 70,-1
{71} 71,4
{72} 72,-1
{73} 73,-1
{74} 74,-1
{75} 75,4
{76} 76,-1
{77} 77,-1
{78} 78,-1
{79} 79,4
{80} 80,-1
{81} 81,-1
{82} 82,-1
{83} 83,4
{84} 84,-1
{85} 85,-1
{86} 86,-1
{87} 87,4
{88} 88,-1
{89} 89,-1
{90} 90,-1
{91} 91,4
{92} 92,-1
{93} 93,-1
{94} 94,-1
{95} 95,4
{96} 96,-1
{97} 97,-1
{98} 98,-1
{99} 99,4
{100} 100,-1

SCIDB QUERY : <store(ARRAY3,ARRAY1)>
[An error expected at this place for the query "store(ARRAY3,ARRAY1)". And it failed with error code = scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR. Expected error code = scidb::SCIDB_SE_INJECTED_ERROR::SCIDB_LE_INJECTED_ERROR.]

SCIDB QUERY : <aggregate(filter(apply(join(ARRAY1,ARRAY2),attrSum,ARRAY1.v + ARRAY2.v), attrSum<>0),count(*))>
{i} count
{0} 0

SCIDB QUERY : <remove(ARRAY1)>
Query was executed successfully

SCIDB QUERY : <remove(ARRAY2)>
Query was executed successfully

SCIDB QUERY : <remove(ARRAY3)>
Query was executed successfully

